### PR TITLE
chore(python): update required checks to include owlbot and GH actions

### PR DESCRIPTION
**Pull request by parthea**
Wait for https://github.com/googleapis/synthtool/pull/1333

chore(python): grant python-samples-reviewers write access to python repos

### CODE REVIEW

**Review by busunkim96 — APPROVED** [review]
LGTM, please make sure we aren't breaking any repos that need to keep Kokoro around for system tests.

File: packages/sync-repo-settings/src/required-checks.json [hunk 1]
```diff
@@ -42,8 +42,17 @@
         "requiresCodeOwnerReviews": true,
         "requiresStrictStatusChecks": true,
         "requiredStatusCheckContexts": [
-          "Kokoro",
```
Comment (tswast) [inline packages/sync-repo-settings/src/required-checks.json:45]: We probably need to wait until system tests are migrated before removing this, unless there are few enough repos that it'd make sense to have a custom `sync-repo-settings.yaml` in those repos temporarily?

**Review by tswast — CHANGES_REQUESTED** [review]
LGTM, but curious how you're handling system tests in the meantime?

Just mailed https://github.com/googleapis/python-db-dtypes-pandas/pull/61 to test out this status checks list in a repo with no system tests.

**Review by tswast — APPROVED** [review]
As discussed in chat, we'll use `sync-repo-settings.yaml` in repos with system tests.

**Review by chingor13 — APPROVED** [review]

**Comment by tmatsuo** [comment]
@parthea Is it ready to merge?

**Pull request merged**